### DISCUSSION TITLE
Invitation email to create account with fixed email

### DISCUSSION THREAD

**Discussion by ArthurJam** [keycloak/keycloak#21477] [Q&A]
Could you please explain how to obtain a registration link for a Keycloak account by forcing the use of the user's email?
I need to create a registration process where users can click on a specific link to activate their account. I would like to avoid users providing a username and simply use their email address as the unique identifier.

Any help would be greatly appreciated. Thank you!

**Comment by sohaibaumer** [comment]
- Configure the Keycloak realm: In the Keycloak administration console, navigate to the desired realm and go to the "Authentication" tab. Under the "Flows" section, select the desired flow (e.g., "Registration") and edit it.
- Configure the registration form: In the flow, ensure that the "Username" field is not required or disabled, and add the "Email" field as a required field. This will force users to provide their email address during registration.
- Obtain the registration link: After the user submits the registration form, Keycloak will automatically send a confirmation email to the provided email address. The email will contain a registration link that the user needs to click to activate their account.
- Customize the email template (optional): If desired, you can customize the email template sent by Keycloak to match your branding and provide clear instructions for the user to click the registration link.
